server.js

```diff
@@ -1,23 +1,98 @@
-"use strict";
+const {
+  ConfigManager: { getConfig, loadSettings, transformObjectKeys },
+  Logger,
+  WebUtils,
+  ServiceStateManager,
+} = require('@dojot/microservice-sdk');
+const camelCase = require('lodash.camelcase');
+const { killApplication } = require('./app/Utils');
+const { CronManager } = require('./app/cron');
 
-const cron = require('./cron');
-const api = require('./api');
-const config = require('./config')
-const healthcheck = require('./healthcheck');
-const logger = require("@dojot/dojot-module-logger").logger;
+const userConfigFile =
+  process.env.CRON_APP_USER_CONFIG_FILE || 'production.conf';
+loadSettings('CRON', userConfigFile);
+const config = getConfig('CRON');
 
-logger.setLevel(config.logger.level);
+// Logger configuration
+Logger.setVerbose(config.logger.verbose);
+Logger.setTransport('console', { level: config.logger['console.level'] });
+if (config.logger['file.enable']) {
+  Logger.setTransport('file', {
+    level: config.logger['file.level'],
+    filename: config.logger['file.filename'],
+  });
+}
+const logger = new Logger('server');
+
+const serviceStateManager = new ServiceStateManager({
+  lightship: transformObjectKeys(config.lightship, camelCase),
+});
 
 process.on('unhandledRejection', (reason) => {
-    logger.error(`Unhandled Rejection at: ${reason.stack || reason}. Bailing out!!`);
-    process.kill(process.pid, "SIGTERM");
+  logger.error(
+    `Unhandled Rejection at: ${reason.stack || reason}. Bailing out!!`
+  );
+  killApplication();
 });
 
-var cronManager = new cron.CronManager();
-cronManager.init().then(() => {
-    healthcheck.init(cronManager);
-    api.init(cronManager, healthcheck.get());
-}).catch(error => {
-    logger.error(`Cron service initialization failed (${error}). Bailing out!!`);
-    process.kill(process.pid, "SIGTERM");
+// Registering the services, shutdown handlers and health checkers
+serviceStateManager.registerService('kafka-consumer');
+serviceStateManager.registerService('kafka-producer');
+serviceStateManager.registerService('db-cron');
+serviceStateManager.registerService('server');
+
+const cronManager = new CronManager(serviceStateManager);
+
+const routes = require('./app/api')(cronManager, Logger);
+
+// create an instance of HTTP server
+const server = WebUtils.createServer({ logger });
+
+const {
+  tokenParsingInterceptor,
```

Comment:
When we change to use the keycloak, you need to change to `tokenParsingKeycloakInterceptor`